src/app/donation-start/donation-start-form/donation-start-form.component.html

```diff
@@ -632,9 +632,14 @@ <h3 class="b-rt-0 b-m-0 b-bold"><span class="span-hr"></span>How does Big Give u
                 Donate now
               </button>
             }
+            <div aria-live="polite">
+              @if (runningFinalPreSubmitUpdate) {
```

Comment:
This spinner replaces the payment button _before_ it is pressed so we know we are ready to ask Stripe to confirm almost immediately. Typically it should show before the button for less than a second in Production.

Reply:
Should be fine, as on previous comment maybe make it show if `! finalPreSubmitComplete` instead of if `runningFinalPreSubmitUpdate` . The donor isn't really interested in what we're running, they only care if we're ready for them to click.

But another thought is to avoid the distracting spinner we could show the button optimistically expecting that the donor won't click it during that one second. Probably not a good experience for donors who do click quickly and we'd have to give an error message to saying they were too quick.

Reply:
I don't think the last option is a good one when times are around a second and there is nothing else required on the step.

Reply:
Even more robust might be if there's a way to key it to the state of the form - maybe if we can get a hash of it or something - and make the confirm button only available if the final submit has been run.

Or we just start with final `finalPreSubmitComplete` = false, also set it to false any time the form is updated, e.g. `this.donationForm.valueChanges.subscribe(()=> this.finalPreSubmitComplete = false);` (although I know calling subscribe is discouraged) and set it to true only on the 2xx from matchbot.

Reply:
on the other hand the most critical part of the donation is the amount which is immutable anyway, so we don't have a risk of taking the wrong amount by not tracking this. Maybe we don't need to be that robust.